src/framework/theme/components/select/option-group.component.ts

```diff
@@ -26,13 +40,49 @@ export class NbOptionGroupComponent {
   }
   set disabled(value: boolean) {
     this._disabled = convertToBoolProperty(value);
+
+    if (this.options) {
+      this.updateOptionsDisabledState();
+    }
   }
   protected _disabled: boolean = false;
 
   @HostBinding('attr.disabled')
   get disabledAttribute(): '' | null {
     return this.disabled ? '' : null;
   }
+
+  @ContentChildren(NbOptionComponent, { descendants: true }) options: QueryList<NbOptionComponent<any>>;
+
+  ngAfterContentInit() {
+    if (this.options.length) {
+      this.asyncUpdateOptionsDisabledState();
+    }
+
+    this.options.changes
+      .pipe(takeWhile(() => this.alive))
+      .subscribe(() => this.asyncUpdateOptionsDisabledState());
+  }
+
+  ngOnDestroy() {
+    this.alive = false;
+  }
+
+  /**
+   * Sets disabled state for each option to current group disabled state.
+   */
+  protected updateOptionsDisabledState(): void {
+    this.options.forEach((option: NbOptionComponent<any>) => option.setDisabledByGroupState(this.disabled));
+  }
+
+  /**
+   * Updates options disabled state after promise resolution.
+   * This way change detection will be triggered after options state updated.
+   * Use this method when updating options during change detection run (e.g. QueryList.changes, lifecycle hooks).
+   */
+  protected asyncUpdateOptionsDisabledState(): void {
```

Comment:
I don't think we have to use `void` return statements :) Because we're not using it always. And I personally hate that thing 😈

Reply:
It will prevent you from returning values if you try to. 🙂